### PR TITLE
Added engraving items factory 

### PR DESCRIPTION
**Pull request by igorkorsukov**

### CODE REVIEW

**Review by Eism — APPROVED** [review]

File: src/engraving/libmscore/engravingobject.cpp [hunk 1]
```diff
@@ -102,6 +107,26 @@ EngravingObject::~EngravingObject()
...
 }
 
+void EngravingObject::doSetParent(EngravingObject* p)
+{
+    if (m_parent == p) {
+        return;
+    }
+
+    if (p) {
+        p->m_onDestroyed.resetOnReceive(this);
```
Comment (cbjeukendrup) [inline src/engraving/libmscore/engravingobject.cpp:117]: Why is this needed? Shouldn't this be `m_parent` (the old parent) instead of `p` (the new one)?

Comment (igorkorsukov) [reply src/engraving/libmscore/engravingobject.cpp:117]: oops, yeah, you're right.
in any case, this is a temporary hack, I will remove this code later.
